hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/test/java/org/apache/hadoop/yarn/server/router/clientrm/TestableFederationClientInterceptor.java

```diff
@@ -90,6 +95,7 @@ protected ApplicationClientProtocol getClientRMProxyForSubCluster(
         mockRMs.put(subClusterId, mockRM);
       }
       initNodeAttributes(subClusterId, mockRM);
+      initReservationSystem(mockRM);
```

Comment:
This seems out of scope.
Should we do it in a separate PR?

Reply:
I will submit pr separately to improve this code.

Reply:
The new pr will optimize the repetitive submission of reservations and refactor some test code.